hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/detailslayout/AbstractSoftwareModuleDetails.java

```diff
@@ -0,0 +1,206 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.ui.common.detailslayout;
+
+import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.hawkbit.repository.EntityFactory;
+import org.eclipse.hawkbit.repository.SoftwareManagement;
+import org.eclipse.hawkbit.repository.model.MetaData;
+import org.eclipse.hawkbit.repository.model.SoftwareModule;
+import org.eclipse.hawkbit.ui.SpPermissionChecker;
+import org.eclipse.hawkbit.ui.artifacts.event.SoftwareModuleEvent;
+import org.eclipse.hawkbit.ui.artifacts.smtable.SoftwareModuleAddUpdateWindow;
+import org.eclipse.hawkbit.ui.components.SPUIComponentProvider;
+import org.eclipse.hawkbit.ui.distributions.event.MetadataEvent;
+import org.eclipse.hawkbit.ui.distributions.smtable.SwMetadataPopupLayout;
+import org.eclipse.hawkbit.ui.management.state.ManagementUIState;
+import org.eclipse.hawkbit.ui.utils.HawkbitCommonUtil;
+import org.eclipse.hawkbit.ui.utils.I18N;
+import org.eclipse.hawkbit.ui.utils.UIComponentIdProvider;
+import org.vaadin.spring.events.EventBus.UIEventBus;
+import org.vaadin.spring.events.EventScope;
+import org.vaadin.spring.events.annotation.EventBusListenerMethod;
+
+import com.vaadin.ui.Button.ClickEvent;
+import com.vaadin.ui.Label;
+import com.vaadin.ui.TabSheet;
+import com.vaadin.ui.UI;
+import com.vaadin.ui.VerticalLayout;
+import com.vaadin.ui.Window;
+
+/**
+ * Abstract class which contains common code for Software Module Details
+ *
+ */
+public abstract class AbstractSoftwareModuleDetails
+        extends AbstractNamedVersionedEntityTableDetailsLayout<SoftwareModule> {
+
+    private static final long serialVersionUID = 1L;
+
+    private final SoftwareModuleMetadatadetailslayout swmMetadataTable;
+
+    private final SoftwareModuleAddUpdateWindow softwareModuleAddUpdateWindow;
+
+    private final transient SoftwareManagement softwareManagement;
+
+    private final SwMetadataPopupLayout swMetadataPopupLayout;
+
+    protected AbstractSoftwareModuleDetails(final I18N i18n, final UIEventBus eventBus,
+            final SpPermissionChecker permissionChecker, final ManagementUIState managementUIState,
+            final SoftwareManagement softwareManagement, final SwMetadataPopupLayout swMetadataPopupLayout,
+            final EntityFactory entityFactory, final SoftwareModuleAddUpdateWindow softwareModuleAddUpdateWindow) {
+        super(i18n, eventBus, permissionChecker, managementUIState);
+        this.softwareModuleAddUpdateWindow = softwareModuleAddUpdateWindow;
+        this.softwareManagement = softwareManagement;
+        this.swMetadataPopupLayout = swMetadataPopupLayout;
+
+        swmMetadataTable = new SoftwareModuleMetadatadetailslayout();
+        swmMetadataTable.init(getI18n(), getPermissionChecker(), softwareManagement, swMetadataPopupLayout,
+                entityFactory);
+
+        addTabs(detailsTab);
```

Comment:
Constructor should not call overridables see: http://www.javapractices.com/topic/TopicAction.do?Id=215